### PR TITLE
BB-3058: Move common terraform files to public repo

### PR DESCRIPTION
**Pull request by giovannicimolin**
This PR moves common terraform configuration to a public repo.
I couldn't just cherry pick terraform commits from the other repo to keep attribution since they're all mixed up with secret variables.

**Reviewers:**
- [x] @pcockwell 
- [x] @bradenmacdonald 

@bradenmacdonald Can you do a review pass before I make this repo and PR public?

### CODE REVIEW

**Comment by bradenmacdonald** [comment]
Nice, thank you @giovannicimolin !

I'd like to include some notes in a README about the remaining TODOs:
- `rds` and `s3` shouldn't be separate modules and should be fully merged into `application`
- Currently this has a few variables that are configurable which don't need to be, so they can be simplified
- We can rename the `application` to `openedx` and move `wordpress` to outside of the `openedx` directory ?
- Add a way to manage domain names and HTTPS certificates with Terraform (create a hosted zone in Route53)
- Create a VPC automatically ?

Could you please add a README.md, and include that? Then good to go. 👍 

:+1:

- [x] I tested this: n/a
- [x] I read through the code
- [x] I checked for accessibility issues: n/a
- [ ] Includes documentation - except readme

**Review by pcockwell — APPROVED** [review]
:+1: Overall this looks decent. I do think a todo list in a README as suggested by @bradenmacdonald is a good idea though. It is important to note that there are minor changes to the KKUx code in an [open PR](https://gitlab.com/opencraft/client/kkux/configuration-secure/-/merge_requests/10/diffs) that might be worth trying to include (only things in the  `common` folder)

- [x] I tested this: n/a
- [x] I read through the code
- [x] I checked for accessibility issues: n/a
- [ ] Includes documentation

**Comment by giovannicimolin** [comment]
@pcockwell Since you did most of the work here, can you rebase and cherry pick the customizations from the unmerged production PR? It'll rename the commit author and keep your open source work visible :)

**Comment by giovannicimolin** [comment]
This repo is now public.

File: modules/openedx/application/deprecated.tf [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+  director_default_sg_description = "Security group for the ${title(var.environment)} director instance"
+  director_security_group_description = coalesce(
+    var.deprecated_director_security_group_description,
+    local.director_default_sg_description
+  )
+
+  memcached_cluster_id = coalesce(
+    var.deprecated_memcached_cluster_id,
+    lower(join("-", [var.client_shortname, "memcached_", var.environment]))
```
Comment (giovannicimolin) [inline modules/openedx/application/deprecated.tf]: Issue with string naming: underline (`_`) is not allowed.

**Pull request merged**